File: src/day_1.c
#include <stdio.h>

int main() {
    FILE* fp = fopen("dat/day_1.txt", "r");
    bool verbose = true;
    int i = 0;
    int j = 0;
    int k;
    int part_1 = 0;
    int part_2 = 0;
    int rem;
    fscanf(fp, "%d", &i);
    while (! feof (fp) && (j < 1000)) {
        if (verbose) {
            printf("%d %d %d\n", j, i, i / 3 - 2);
        }

        rem = i / 3 - 2;
        k = 0;
        while (rem > 0 && k < 1000) {
                if (verbose) {
                    printf("    %d %d\n", part_2, rem);
                }
                part_2 += rem;
                rem = rem / 3 - 2;
                k++;
            }
        part_1 += i / 3 - 2;
        fscanf(fp, "%d", &i);
        j++;
    }
    printf("Part 1 Fuel: %d\n", part_1);

    j = 0;
    while (rem > 0 && j < 1000) {
        if (verbose) {
            printf("%d %d\n", part_2, rem);
        }
        part_2 += rem;
        rem = rem / 3 - 2;
        j++;
    }
    printf("Part 2 Fuel: %d\n", part_2);
    return 0;
}

File: src/day_1.jl
#!/usr/bin/env julia

using Printf
using ArgParse
using Dates
using JSON
using ProgressBars


function parse_cmdline()
    parser = ArgParseSettings()

    @add_arg_table! parser begin
        "-v", "--verbose"
            help="Verbose debugging"
            action = :count_invocations
        "-t", "--test"
            help="Use test file"
            action = :store_true
        # "count"
            # arg_type=Int
            # required = true
    end
    return parse_args(parser)
end


function main()
    args = parse_cmdline()
    path_parts = split(PROGRAM_FILE, "/")
    day = split(split(path_parts[length(path_parts)], "_")[2], ".")[1]
    project_dir = dirname(dirname(abspath(PROGRAM_FILE)))
    if !args["test"]
        data_path = joinpath(project_dir, "dat/day_" * day * ".txt")
    else
        data_path = joinpath(project_dir, "dat/day_" * day * "_test.txt")
    end
    
    inputs = readlines(open(data_path, "r"))
    depths = Array{Int, 1}(undef, length(inputs))
    for i=1:length(depths)
        depths[i] = parse(Int, inputs[i])
    end

    prev = 1000000
    part_1 = 0
    for d in depths
        if d > prev
            part_1 += 1
        end
        prev = d
    end

    @printf("Part 1: %d\n", part_1)
    part_2 = 0
    prev = 1000000
    for i=1:length(depths) - 2
        # low = i > 2 ? i - 2 : 1
        # high = i < length(depths) ? i : length(depths)
        window = depths[i:i+2]
        cur = sum(window)
        if cur > prev
            part_2 += 1
        end
        prev = cur
    end

    @printf("Part 2: %d\n", part_2)

end

main()
